src/main/java/ch/njol/skript/expressions/ExprJoinSplit.java

```diff
@@ -82,12 +64,15 @@ public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelaye
 	protected String[] get(Event event) {
 		String[] strings = this.strings.getArray(event);
 		String delimiter = this.delimiter != null ? this.delimiter.getSingle(event) : "";
+
 		if (strings.length == 0 || delimiter == null)
 			return new String[0];
+
 		if (join) {
-			return new String[] {StringUtils.join(strings, delimiter)};
+			return new String[]{StringUtils.join(strings, delimiter)};
 		} else {
-			return strings[0].split(regex ? delimiter : (caseSensitivity ? "" : "(?i)") + Pattern.quote(delimiter), -1);
+			return strings[0].split(regex ? delimiter : (caseSensitivity ? "" : "(?i)") + Pattern.quote(delimiter),
```

Comment:
The method call is getting long, reassign `delimiter` above it if `regex` is false